### PR TITLE
Docs homepage

### LINKED ISSUE #3900: Eliminate redirect on initial load of docs.flutter.io
docs.flutter.io redirects to docs.flutter.io/flutter.

I think, now that we have a unified API doc set, we can eliminate this redirect and speed up the initial load of our site.

Bonus, this helps verify the site in Webmaster Tools :)

### PR DESCRIPTION
**Pull request by sethladd**
Closes https://github.com/flutter/flutter/issues/3900

### CODE REVIEW

**Comment by sethladd** [comment]
cc @abarth @devoncarew

**Comment by devoncarew** [comment]
lgtm

**Comment by sethladd** [comment]
Thanks! I'll merge Monday, just in case.

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -8,9 +8,16 @@ import 'dart:io';
 
 import 'package:path/path.dart' as path;
 
+const String DOC_ROOT = 'dev/docs/doc';
```
Comment (devoncarew) [inline dev/tools/dartdoc.dart]: ==> `kDocRoot`

File: dev/tools/dartdoc.dart [hunk 2]
```diff
@@ -56,24 +63,61 @@ dependencies:
...
+void createIndexAndCleanup() {
+  print('\nCreating a custom index.html in $DOC_ROOT/index.html');
+  renameApiDir();
+  copyIndexToRootOfDocs();
+  addHtmlBaseToIndex();
+  putRedirectInOldIndexLocation();
+  print('\nDocs ready to go!');
+}
+
+Directory renameApiDir() {
```
Comment (devoncarew) [inline dev/tools/dartdoc.dart]: We're not using the return values of `renameApiDir` (or `copyIndexToRootOfDocs` below).

Comment (sethladd) [inline dev/tools/dartdoc.dart]: Changed to void.

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (sethladd) [inline dev/tools/dartdoc.dart]: changed

**Pull request merged**